src/pages/SolutionAnalysisPreview.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useState, useEffect } from 'react';
+import { LatLngExpression } from 'leaflet';
+import 'leaflet/dist/leaflet.css';
+import { MapContainer, TileLayer, GeoJSON } from 'react-leaflet';
+import axios from 'axios';
+import SelectControl from '../components/common/SelectControl';
+import { Alert, Button, Card, Typography } from '@mui/material';
+import { styled } from '@mui/styles';
+import MultiSelect from '../components/common/MultiSelect';
+import { GeoJsonObject } from 'geojson';
+
+const FloatingCard = styled(Card)({
+  zIndex: 401,
+  padding: 20,
+  disply: 'flex',
+  justifyContent: 'center',
+  left: '3%',
+  top: '10%',
+});
+
+const ControlsBar = styled('div')({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const myStyle = {
+  color: '#000000',
+  weight: 1,
+  opacity: 0.65,
+};
+interface LocationData {
+  id: string;
+  latitude: number;
+  longitude: number;
+  name: string;
+  population: number;
+}
+const solvisEndpoint = process.env.REACT_APP_SOLVIS_ENDPOINT as string;
+
+const SolutionAnalysisPreview: React.FC = () => {
+  const [locationOptions, setLocationOptions] = useState<LocationData[]>([]);
+  const [locationSelections, setLocationSelections] = useState<string[]>([]);
+  const [locationIDs, setLocationIDs] = useState<string[]>([]);
+
+  const [radiiOptions, setRadiiOptions] = useState<string[]>([]);
+  const [radiiSelection, setRadiiSelection] = useState<string>('');
+
+  const [geojsonData, setGeoJsonData] = useState<GeoJsonObject>();
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const nz_centre: LatLngExpression = [-40.946, 174.167];
+  const zoom = 5;
+
+  const provider_url = 'https://server.arcgisonline.com/ArcGIS/rest/services/Ocean_Basemap/MapServer/tile/{z}/{y}/{x}';
+
+  useEffect(() => {
+    const filteredLocations = locationOptions.filter((location) => locationSelections.includes(location.name));
+    const filteredLocationIDs: string[] = [];
+    filteredLocations.map((location) => {
+      filteredLocationIDs.push(location.id);
+    });
+    setLocationIDs(filteredLocationIDs);
+  }, [locationSelections]);
+
+  useEffect(() => {
+    const locationListID = process.env.REACT_APP_ANALYSIS_LOC_LIST_ID as string;
+    const radiiID = process.env.REACT_APP_RADII_ID;
+
+    axios
```

Comment:
Interested in the use of `axios` here - I believe we've already used `fetch` for our purposes, is there a reason for the new library?

Reply:
Do you think I should use fetch instead for these api calls and no use a new library?

Reply:
No I think it should be fine, was just curious.